src/main/java/uk/gov/register/views/EntryListView.java

```diff
@@ -1,36 +1,109 @@
 
+import javax.inject.Provider;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
 import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
 public class EntryListView implements CsvRepresentationView {
-    private Collection<Entry> entries;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EntryListView.class);
+
+    private static final String END_OF_LINE = "\n";
+
+    private final Collection<Entry> entries;
     private final Optional<String> recordKey;
 
-    public EntryListView(Collection<Entry> entries) {
+    private final ObjectMapper jsonObjectMapper = Jackson.newObjectMapper();
+    private final ObjectMapper yamlObjectMapper = Jackson.newObjectMapper(new YAMLFactory());
+
+    public EntryListView(final Collection<Entry> entries) {
         this.entries = entries;
-        this.recordKey = Optional.empty();
+        recordKey = Optional.empty();
     }
 
-    public EntryListView(Collection<Entry> entries, String recordKey) {
+    public EntryListView(final Collection<Entry> entries, final String recordKey) {
         this.entries = entries;
         this.recordKey = Optional.of(recordKey);
     }
 
+    public String entriesTo(final String mediaType, final Provider<RegisterName> registerNameProvider) {
+        final ByteArrayOutputStream outputStream;
+        final RegisterResolver registerResolver;
+        final EntryListTurtleWriter entryTurtleWriter;
+        String registerInTextFormatted = StringUtils.EMPTY;
+
+        try {
+            if (ExtraMediaType.TEXT_TTL_TYPE.getSubtype().equals(mediaType)) {
+                outputStream = new ByteArrayOutputStream();
+                registerResolver = registerName -> URI.create("http://" + registerName + ".register.gov.uk");
+                entryTurtleWriter = new EntryListTurtleWriter(registerNameProvider, registerResolver);
+
+                entryTurtleWriter.writeTo(this, EntryListView.class, EntryListView.class, null, ExtraMediaType.TEXT_TTL_TYPE, null, outputStream);
+
+                registerInTextFormatted = new String(outputStream.toByteArray(), StandardCharsets.UTF_8);
+            } else if (ExtraMediaType.TEXT_YAML_TYPE.getSubtype().equals(mediaType)) {
+                registerInTextFormatted = yamlObjectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(getNestedEntryJson());
+            } else {
+                registerInTextFormatted = jsonObjectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(getNestedEntryJson());
+            }
+        } catch (final IOException ex) {
+            LOGGER.error("Error processing preview request. Impossible process the register items");
+        }
+
+        return StringEscapeUtils.escapeHtml(registerInTextFormatted.isEmpty() ? registerInTextFormatted : END_OF_LINE + registerInTextFormatted);
```

Comment:
Two questions: why `\n` if `registerInTextFormatted` is not empty (as opposed to the view taking care of this) and do we notify the user there was an error if there was an error?

Reply:
Discussed.